pymysql/connections.py

```diff
@@ -805,6 +805,7 @@ def _request_authentication(self):
         plugin_name = None
 
         if self._auth_plugin_name in ('', 'mysql_native_password'):
+            plugin_name = b'mysql_native_password'
             authresp = _auth.scramble_native_password(self.password, self.salt)
```

Comment:
When server sent `b''`, shouldn't we send `b''`?

Reply:
You are right, let me fix it.

Reply:
Hi methane, 
I do not think `When server sent b'', shouldn't we send b''?`. That is not correct.
The plugin_name should be consist with the `authresp` field, when response the plugin_name, it tells the server `authresp` method.

Reply:
Both of `b''` and `b'mysql_native_password'` are compatible with authresp, because both are same implementation.  Am I wrong?

>  The plugin_name should be consist with the authresp field, when response the plugin_name, it tells the server authresp method.

AFAIK, plugin_name in handshake response must be consistent with plugin_name server sent.  And authresp must consistent with plugin_name.  If plugin_name is not same to server sent, server may request `Auth Change`, without reading authresp.

Could you setup MySQL and MariaDB server which supports plugin, but send `b''` as plugin name on initial packet,
and capture all packets?
(Yes, developing auth flow is very bothersome job.  But even you think "it works", initial handshake may be broken and auth is succeeded after "Auth Switch")

Reply:
Hi methane,
You are right. 
I think it too simple. After I tested MariaDB Embedded server, I find your knowledge and experience is correct. Thanks a lot for correcting me.